avalanche/training/losses.py

```diff
@@ -161,4 +164,61 @@ def forward(self, features, labels=None, mask=None):
         return loss
 
 
-__all__ = ["ICaRLLossPlugin", "SCRLoss"]
+class MaskedCrossEntropy(SupervisedPlugin):
+    """
+    Masked Cross Entropy
+
+    This criterion can be used for instance in Class Incremental
+    Learning Problems when no examplars are used
+    (i.e LwF in Class Incremental Learning would need to use mask="new").
+    """
+
+    def __init__(self, classes=None, mask="all", reduction="mean"):
+        """
+        param: classes: Initial value for current classes
+        param: mask: "all" normal cross entropy, uses all the classes seen so far
+                     "old" cross entropy only on the old classes
+                     "new" cross entropy only on the new classes
+        param: reduction: "mean" or "none", average or per-sample loss
+        """
+        super().__init__()
+        assert mask in ["all", "new", "old"]
+        if classes is not None:
+            self.current_classes = set(classes)
+        else:
+            self.current_classes = set()
+
+        self.old_classes = set()
+        self.reduction = reduction
+        self.mask = mask
+
+    def __call__(self, logits, targets):
+        oh_targets = F.one_hot(targets, num_classes=logits.shape[1])
+
+        oh_targets = oh_targets[:, self.current_mask]
+        logits = logits[:, self.current_mask]
+
+        return cross_entropy_with_oh_targets(
+            logits,
+            oh_targets.float(),
+            reduction=self.reduction,
+        )
+
+    @property
+    def current_mask(self):
+        if self.mask == "all":
```

Comment:
this is incorrect. You are selecting only "seen" units, which excludes future units (e.g. DER regularizes them).

Reply:
I think it's better to have both "seen" and "all" options (not necessarily for all losses).